templates/commands/render_action.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"bytes"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/abcxyz/abc/templates/model"
+)
+
+// Called with the contents of a file, and returns the new contents of the file
+// to be written.
+type walkAndModifyVisitor func([]byte) ([]byte, error)
+
+// Recursively traverses the directory or file scratchDir/relPath, calling the
+// given visitor for each file. If the visitor returns modified file contents
+// for a given file, that file will be overwritten with the new contents.
+func walkAndModify(pos *model.ConfigPos, rfs renderFS, scratchDir, relPath string, v walkAndModifyVisitor) error {
```

Comment:
Recommend moving a few funcs over `parseAndExecuteGoTmpl` and related, and `copyRecursive` and related.

Reply:
Ok, will do in a followup PR so I don't lose the precious approval bit by adding new commits to this PR.